decoder/decoder.go

```diff
@@ -172,6 +175,17 @@ func countAttributeSchema() *schema.AttributeSchema {
 	}
 }
 
+func forEachAttributeSchema() *schema.AttributeSchema {
+	return &schema.AttributeSchema{
+		IsOptional: true,
+		Expr: schema.ExprConstraints{
+			schema.TraversalExpr{OfType: cty.Map(cty.String)},
+			schema.LiteralTypeExpr{Type: cty.Set(cty.String)},
```

Comment:
Should this match https://github.com/hashicorp/terraform-schema/blob/aac2bff86a0b7014a2d2b70044e6558b2edc3fd6/internal/schema/0.12/resource_block.go#L80-L91?